tiatoolbox/wsicore/wsireader.py

```diff
@@ -984,43 +1128,70 @@ def read_bounds(
         interpolation="optimise",
         pad_mode="constant",
         pad_constant_values=0,
+        location_is_at_requested=False,
         **kwargs,
     ):
-
-        # Find parameters for optimal read
-        (
-            read_level,
-            level_bounds,
-            output_size,
-            post_read_scale,
-        ) = self._find_read_bounds_params(bounds, resolution=resolution, units=units)
+        # convert from requested to `lv0`
+        bounds_at_lv0 = bounds
+        if location_is_at_requested:
+            bounds_at_lv0 = self._bounds_at_requested_to_lv0(bounds, resolution, units)
+            _, size_at_requested = utils.transforms.bounds2locsize(bounds)
+            # dont use the `output_size` (`size_at_requested`) here
+            # because the rounding error at `bounds_at_lv0` leads to
+            # different `size_at_requested` (keeping same read resolution
+            # but base image is of different scale)
+            (
+                read_level,
+                bounds_at_read_lv,
+                _,
+                post_read_scale,
+            ) = self._find_read_bounds_params(
+                bounds_at_lv0, resolution=resolution, units=units
+            )
+        else:  # duplicated portion with VirtualReader, factoring out ?
+            # Find parameters for optimal read
+            (
+                read_level,
+                bounds_at_read_lv,
```

Comment:
Could we use `bounds_at_read_level` here? It's not much longer and matches the previous variable name then. Also, I think it's a bit more clear.